src/testFixtures/java/org/opensearch/knn/KNNRestTestCase.java

```diff
@@ -801,6 +820,129 @@ public void validateKNNSearch(String testIndex, String testField, int dimension,
+                .endObject()
+                .endObject()
+        );
+    }
+
+    // Default KNN script score settings
+    protected Settings getKNNScriptScoreSettings() {
+        return Settings.builder().put(NUMBER_OF_SHARDS, 1).put(NUMBER_OF_REPLICAS, 0).put(INDEX_KNN, false).build();
+    }
+
+    // Validate script score search for these space_types : {"l2", "l1", "linf"}
+    protected void validateKNNScriptScoreSearch(String testIndex, String testField, int dimension, int numDocs, int k, SpaceType spaceType)
+        throws Exception {
+        float[] queryVector = new float[dimension];
+        Arrays.fill(queryVector, (float) numDocs);
+
+        QueryBuilder qb = new MatchAllQueryBuilder();
+        Map<String, Object> params = new HashMap<>();
+        params.put(FIELD, testField);
+        params.put(QUERY_VALUE, queryVector);
+        params.put(METHOD_PARAMETER_SPACE_TYPE, spaceType.getValue());
+
+        Request request = constructKNNScriptQueryRequest(testIndex, qb, params, k);
+        Response response = client().performRequest(request);
+        assertEquals(request.getEndpoint() + ": failed", RestStatus.OK, RestStatus.fromCode(response.getStatusLine().getStatusCode()));
+
+        List<KNNResult> results = parseSearchResponse(EntityUtils.toString(response.getEntity()), testField);
+        assertEquals(k, results.size());
+
+        PriorityQueue<DistVector> pq = computeGroundTruthValues(queryVector, numDocs, k, dimension, spaceType);
+
+        for (int i = k - 1; i >= 0; i--) {
+            int expDocID = Integer.parseInt(pq.poll().getDocID());
+            int actualDocID = Integer.parseInt(results.get(i).getDocId());
+            assertEquals(expDocID, actualDocID);
+        }
+    }
+
+    // validate KNN painless script score search for the space_types : "l2", "l1"
+    protected void validateKNNPainlessScriptScoreSearch(String testIndex, String testField, String source, int numDocs, int k)
+        throws Exception {
+        QueryBuilder qb = new MatchAllQueryBuilder();
+        Request request = constructScriptQueryRequest(testIndex, qb, Collections.emptyMap(), Script.DEFAULT_SCRIPT_LANG, source, k);
+        Response response = client().performRequest(request);
+        assertEquals(request.getEndpoint() + ": failed", RestStatus.OK, RestStatus.fromCode(response.getStatusLine().getStatusCode()));
+
+        List<KNNResult> results = parseSearchResponse(EntityUtils.toString(response.getEntity()), testField);
+        assertEquals(k, results.size());
+
+        for (int i = 0; i < k; i++) {
+            int expDocID = numDocs - i - 1;
+            int actualDocID = Integer.parseInt(results.get(i).getDocId());
+            assertEquals(expDocID, actualDocID);
+        }
+    }
+
+    // generate painless script source for space_type "l2" by creating query vector based on number of documents
+    protected String generateL2PainlessScriptSource(String testField, int dimension, int numDocs) {
```

Comment:
Instead of generate, can we rename to create to keep consistent with some above methods?